### PR TITLE
Use IR nodes for casts, binary operations, unary operations 

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
@Kixiron please check whether this addresses your requests

**Comment by mbudiu-vmw** [comment]
@lalithsuresh this seems to also have run 4 CI tests

**Comment by Kixiron** [comment]
What's the output json look like?
Also, I have support for unsigned integers if you want to add those to your frontend (I think it's relatively standard with `UNSIGNED`)

**Comment by mbudiu-vmw** [comment]
Unfortunately SQL has no support for unsigned integers, so I don't know how to generate properly such code.
I think you should try to build the compiler, it's really one command-line you need to run if you have installed java and maven. Then you can generate lots of json easily - one more command-line.

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/DBSPCompilerTests.java [hunk 1]
```diff
@@ -65,6 +64,7 @@ public void circuitToJsonTest() throws SqlParseException, JsonProcessingExceptio
         compiler.compileStatement("CREATE VIEW V AS SELECT * FROM T WHERE COL1 > 5");
         DBSPCircuit dbsp = compiler.getFinalCircuit("circuit");
         String json = ToJSONVisitor.circuitToJSON(dbsp);
+        System.out.println(json);
```
Comment (lalithsuresh) [inline SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/DBSPCompilerTests.java:67]: Remove

File: doc/sql/boolean.rst [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    - NULL
+  * - NULL
+    - TRUE
+    - NULL
+    - NULL
+
+AND
+---
+
+The following truth table defines the ``OR`` operator:
```
Comment (lalithsuresh) [inline doc/sql/boolean.rst:35]: ``AND``

File: doc/sql/comparisons.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  * - ``<``
+    - less than
+    -
+  * - ``>=``
+    - greater or equal
+    -
+  * -  ``<=``
+    - less or equal
+    -
+  * - `IS NULL``
```
Comment (lalithsuresh) [inline doc/sql/comparisons.rst:31]: Extra backtick at the end.

File: doc/sql/integer.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
+Integer Operations
+==================
+
+There are four supported integer datatypes, ``TINYINT``, ``SMALLINT``,
```
Comment (lalithsuresh) [inline doc/sql/integer.rst:4]: Can we also state (if possible) how many bits each integer type has?

**Review by lalithsuresh — CHANGES_REQUESTED** [review]
Just some nits.

File: doc/sql/integer.rst (hunk 4 above)
Comment (mbudiu-vmw) [reply doc/sql/integer.rst:4]: That is in the section on types. But I can repeat that.

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**